### PR TITLE
Processing data in models

### PR DESCRIPTION
**Pull request by ForHemer**
I set up models for the **Blog App** application in this milestone. I've :

- [ ] Created model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
- [ ] Set up associations between models.
- [ ] Created custom methods.
- [ ] Used models to insert data.
- [ ] Made current user data available to all controllers.

For more details, see [project requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md).

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,2 +1,20 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  after_initialize :increment_by_one
+  after_destroy :decrement_by_one
+
+  def increment_by_one
+    author.increment! :PostCounter
```
Comment (Deepakdanger) [inline app/models/post.rb]: - [ ] Here typo error, pls change `PostCounter` to `PostsCounter`.

File: app/models/post.rb [hunk 2]
```diff
@@ -1,2 +1,20 @@
...
+  end
+
+  def decrement_by_one
+    author.decrement! :PostCounter
```
Comment (Deepakdanger) [inline app/models/post.rb]: - [ ] Here typo error, pls change `PostCounter` to `PostsCounter`.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @ForHemer, 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by t-yanick — APPROVED** [review]
Hi @ForHemer, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**